project-base/storefront/components/Basic/ModalGallery/ModalGallery.tsx

```diff
@@ -0,0 +1,212 @@
+import { PlayIcon } from 'components/Basic/Icon/IconsSvg';
+import { Image } from 'components/Basic/Image/Image';
+import { VideoTokenFragmentApi, ImageFragmentApi } from 'graphql/generated';
+import { twMergeCustom } from 'helpers/twMerge';
+import { useKeypress } from 'hooks/useKeyPress';
+import useTranslation from 'next-translate/useTranslation';
+import { RefObject, createRef, useEffect, useState } from 'react';
+import { useSwipeable } from 'react-swipeable';
+import { twJoin } from 'tailwind-merge';
+
+type ModalGalleryProps = {
+    galleryName: string;
+    initialIndex: number;
+    items: (VideoTokenFragmentApi | ImageFragmentApi)[];
+    onCloseModal: () => void;
+};
+
+export const ModalGallery: FC<ModalGalleryProps> = ({ initialIndex, items, galleryName, onCloseModal }) => {
+    const { t } = useTranslation();
+
+    const [selectedIndex, setSelectedIndex] = useState(initialIndex);
+    const [itemsRefs, setItemsRefs] = useState<Array<RefObject<HTMLLIElement>>>();
+    const [isLoaded, setIsLoaded] = useState(false);
+
+    const selectedGalleryItem = items[selectedIndex];
+
+    const isImage = selectedGalleryItem.__typename === 'Image';
+    const isVideo = selectedGalleryItem.__typename === 'VideoToken';
+
+    const lastItemIndex = items.length - 1;
+
+    const isCarouselDisplayed = items.length > 1 && (!isImage || isLoaded);
+
+    const selectPreviousItem = () => {
+        setSelectedIndex(selectedIndex > 0 ? selectedIndex - 1 : lastItemIndex);
+    };
+
+    const selectNextItem = () => {
+        setSelectedIndex(selectedIndex < lastItemIndex ? selectedIndex + 1 : 0);
+    };
+
+    useEffect(() => {
+        setItemsRefs(
+            Array(items.length)
+                .fill(null)
+                .map(() => createRef()),
+        );
+    }, []);
```

Comment:
Cannot it be initialized with this value, instead of setting it afterwards? What is the reason in this case?

Reply:
Ah, good idea, yes! That should works too.